api/v1alpha1/project_config_types.go

```diff
@@ -46,6 +86,29 @@ type PromotionPolicy struct {
 	AutoPromotionEnabled bool `json:"autoPromotionEnabled,omitempty" protobuf:"varint,2,opt,name=autoPromotionEnabled"`
 }
 
+// WebhookReceiverConfig describes the configuration for webhook receivers.
+type WebhookReceiverConfig struct {
+	// GitHub contains the configuration for a webhook receiver that is compatible with
+	// GitHub payloads.
+	GitHub *GitHubWebhookReceiver `json:"github,omitempty" protobuf:"bytes,1,opt,name=github"`
+}
+
+type GitHubWebhookReceiver struct {
+	// Secret contains a reference to a Secret in the same namespace as the ProjectConfig.
+	//
+	// The Secret is expected to contain a `token` key with the secret token configured for
```

Comment:
fyi, the GitHub docs use the phrase "secret token" or sometimes just "token," but the forms you actually fill out when registering either a GitHub App or an individual webhook say "Secret." Just be aware of this discrepancy when documenting this (which can be in a follow-up).

It might not hurt to add to the comment here that the value of the `token` key goes in the "Secret" field when registering a GitHub App or webhook.